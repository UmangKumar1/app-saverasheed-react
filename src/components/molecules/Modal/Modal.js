import React, { Component } from 'react'
import styled, { css, keyframes, cx } from 'react-emotion'
import PropTypes from 'prop-types'

import ScreenEffect from '../../templates/ScreenEffect'
import theme from '../../../theme'
import { noop } from '../../../util/ramda-extra'

const { bool, any, string, func } = PropTypes

const {
  palette: {
    primary: { main, dark },
  },
} = theme

const ModalRoot = styled('div')(
  {
    position: 'fixed',
    top: 0,
    right: 0,
    bottom: 75,
    left: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: '0px 15px',
    zIndex: 999,
    textAlign: 'center',
    background: 'rgba(0,0,0, 0.7)',
    transition: 'background 300ms',
  },
  ({ on }) => !on && { pointerEvents: 'none', background: 'rgba(0,0,0, 0.0)' },
  ({ noOverlay }) =>
    noOverlay && {
      background: 'rgba(0,0,0, 0.0)',
    },
)

export const ModalTitle = styled('h1')({
  lineHeight: '30px',
  textAlign: 'center',
})

const screenAnim = keyframes`
0% {
  background-position: 0 0;
}

100% {
  background-position: 0 30px;
}
`

const ModalWrapper = styled('div')({
  // background: `linear-gradient(to bottom, ${main}, ${main} 50%, ${dark} 50%, ${dark})`,
  // backgroundSize: '100% 40px',
  background: `linear-gradient(to bottom, ${dark}, ${dark} 90%, ${main} 90%, ${main})`,
  backgroundSize: '100% 30px',
  animation: `${screenAnim} 1.5s linear infinite`,
  padding: 8,
  // height: '100%',
  maxHeight: '100%',
  pointerEvents: 'all',
  overflow: 'scroll',
  borderRadius: 10,
})

const ModalBody = styled('div')({
  padding: '2px 8px 8px',
  backgroundColor: main,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'space-between',
})

export const ModalP = styled('p')({})

const screenEffectCx = css({
  border: 'none',
  background: 'none',
  lineHeight: '20px',
  // height: 'auto',
})

class Modal extends Component {
  static propTypes = {
    on: bool,
    noOverlay: bool,
    children: any,
    wrapperClassName: string,
    bodyClassName: string,
    screenEffectClassName: string,
    onClick: func,
    onOverlayClick: func,
  }

  render() {
    const {
      on,
      children,
      bodyClassName,
      wrapperClassName,
      screenEffectClassName,
      onClick = noop,
      onOverlayClick = noop,
      ...props
    } = this.props
    return (
      <ModalRoot on={on} {...props} onClick={onOverlayClick}>
        <ScreenEffect
          on={on}
          className={cx(screenEffectCx, screenEffectClassName)}
          onClick={e => e.stopPropagation()}
          showAnim
        >
          <ModalWrapper className={wrapperClassName}>
            <ModalBody
              className={bodyClassName}
              onClick={e => {
                e.stopPropagation()
                onClick(e)
              }}
            >
              {children}
            </ModalBody>
          </ModalWrapper>
        </ScreenEffect>
      </ModalRoot>
    )
  }
}

export default Modal
