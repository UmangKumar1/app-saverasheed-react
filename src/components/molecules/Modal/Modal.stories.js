import React, { Component, Fragment } from 'react'
import { storiesOf } from '@storybook/react'
import Modal from '.'

class ModalStoryOn extends Component {
  state = { open: false }

  render() {
    const { open } = this.state
    return (
      <Fragment>
        <button onClick={() => this.setState({ open: true })}>Open!</button>
        <Modal
          isLocationAllowed
          on={open}
          onClick={() => this.setState({ open: false })}
        >
          <div>Testing the modal!</div>
          <div>Testing the modal!</div>
          <div>Testing the modal!</div>
        </Modal>
      </Fragment>
    )
  }
}

class ModalStoryOff extends Component {
  state = { open: true }

  render() {
    const { open } = this.state
    return (
      <Fragment>
        <button onClick={() => this.setState({ open: true })}>Open!</button>
        <Modal
          isLocationAllowed
          on={open}
          onClick={() => this.setState({ open: false })}
        >
          <div>Testing the modal!</div>
          <div>Testing the modal!</div>
          <div>Testing the modal!</div>
        </Modal>
      </Fragment>
    )
  }
}

storiesOf('Modal', module)
  .add('on', () => <ModalStoryOn />)
  .add('off', () => <ModalStoryOff />)
