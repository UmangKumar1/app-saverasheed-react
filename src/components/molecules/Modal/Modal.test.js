import React from 'react'
import { shallow } from 'enzyme'
import Modal from '.'

const wrap = (props = {}) => shallow(<Modal {...props} />)

describe('Modal', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      allowLocation: jest.fn(),
      isLocationAllowed: true,
      on: true,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
