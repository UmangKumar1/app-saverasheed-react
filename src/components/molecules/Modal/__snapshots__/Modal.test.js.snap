// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    allowLocation={[MockFunction]}
    isLocationAllowed={true}
    on={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowLocation": [MockFunction],
      "children": <ScreenEffect
        className="css-1hynkh6"
        on={true}
        onClick={[Function]}
        showAnim={true}
      >
        <Styled(div)>
          <Styled(div)
            onClick={[Function]}
          />
        </Styled(div)>
      </ScreenEffect>,
      "isLocationAllowed": true,
      "on": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Styled(div)>
          <Styled(div)
            onClick={[Function]}
          />
        </Styled(div)>,
        "className": "css-1hynkh6",
        "on": true,
        "onClick": [Function],
        "showAnim": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(div)
            onClick={[Function]}
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowLocation": [MockFunction],
        "children": <ScreenEffect
          className="css-1hynkh6"
          on={true}
          onClick={[Function]}
          showAnim={true}
        >
          <Styled(div)>
            <Styled(div)
              onClick={[Function]}
            />
          </Styled(div)>
        </ScreenEffect>,
        "isLocationAllowed": true,
        "on": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(div)>
            <Styled(div)
              onClick={[Function]}
            />
          </Styled(div)>,
          "className": "css-1hynkh6",
          "on": true,
          "onClick": [Function],
          "showAnim": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(div)
              onClick={[Function]}
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
